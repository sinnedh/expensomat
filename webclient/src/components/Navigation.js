import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const Navigation = (props, context) => {
  return (
    <ul>
      <li><Link to="/calculation">Create new calculation</Link></li>
      <li><Link to={`/calculation/${props.token}`}>Show calculation</Link></li>
      <li><Link to={`/calculation/${props.token}/manage`}>Manage calculation</Link></li>
    </ul>
  )
}

Navigation.contextTypes = {token: PropTypes.string}

export default Navigation
