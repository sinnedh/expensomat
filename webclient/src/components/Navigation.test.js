import React from 'react'
import ReactDOM from 'react-dom'
import { Map } from 'immutable'
import { MemoryRouter } from 'react-router'
import configureStore from 'redux-mock-store'
import Navigation from './Navigation'

const renderNavigation = (user) => (
  shallow(
    <MemoryRouter location="someLocation">
      <Navigation token={"1234"} user={user}/>
    </MemoryRouter>
  ).find('Navigation').dive()
)

describe('Navigation', () => {
  const initialState = Map()
  const mockStore = configureStore()
  let store, container

  beforeEach(() => {
    store = mockStore(initialState)
  })

  it('renders properly without user', () => {
    const wrapper = shallow(
      <MemoryRouter location="someLocation">
        <Navigation token={"1234"} />
      </MemoryRouter>
    )
    expect(renderNavigation(null)).toMatchSnapshot()
  })

  it('renders properly with admin-user', () => {
    const user = {name: 'Keek', role: 'admin'}
    expect(renderNavigation(user)).toMatchSnapshot()
  })

  it('renders properly with editor-user', () => {
    const user = {name: 'Keek', role: 'editor'}
    expect(renderNavigation(user)).toMatchSnapshot()
  })

  it('renders properly with observer-user', () => {
    const user = {name: 'Keek', role: 'observer'}
    expect(renderNavigation(user)).toMatchSnapshot()
  })
})
