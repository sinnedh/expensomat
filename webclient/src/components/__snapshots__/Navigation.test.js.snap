// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders properly with admin-user 1`] = `
<ul>
  <li>
    Hello 
    Keek
    !
  </li>
  <li>
    <Link
      replace={false}
      to="/calculation/1234"
    >
      Show calculation
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/calculation/1234/manage"
    >
      Manage calculation
    </Link>
  </li>
</ul>
`;

exports[`Navigation renders properly with editor-user 1`] = `
<ul>
  <li>
    Hello 
    Keek
    !
  </li>
  <li>
    <Link
      replace={false}
      to="/calculation/1234"
    >
      Show calculation
    </Link>
  </li>
</ul>
`;

exports[`Navigation renders properly with observer-user 1`] = `
<ul>
  <li>
    Hello 
    Keek
    !
  </li>
  <li>
    <Link
      replace={false}
      to="/calculation/1234"
    >
      Show calculation
    </Link>
  </li>
</ul>
`;

exports[`Navigation renders properly without user 1`] = `<ul />`;
